uniffi_macros/src/object.rs

```diff
@@ -67,28 +67,26 @@ pub fn expand_object(input: DeriveInput, options: DeriveOptions) -> syn::Result<
         #[doc(hidden)]
         #[unsafe(no_mangle)]
         pub unsafe extern "C" fn #clone_fn_ident(
-            ptr: *const ::std::ffi::c_void,
+            handle: ::std::primitive::u64,
             call_status: &mut ::uniffi::RustCallStatus
-        ) -> *const ::std::ffi::c_void {
-            ::uniffi::deps::trace!("clone: {} ({:?})", #name, ptr);
+        ) -> ::std::primitive::u64 {
+            ::uniffi::deps::trace!("clone: {} ({:?})", #name, handle);
             ::uniffi::rust_call(call_status, || {
-                unsafe { ::std::sync::Arc::increment_strong_count(ptr) };
-                ::std::result::Result::Ok(ptr)
+                unsafe { ::std::sync::Arc::increment_strong_count(handle as *const #ident) };
```

Comment:
IIRC, we used to need to do this in two steps: `handle as *() as *const #ident`.  I'm not sure when this changed in Rust, but it doesn't seem necessary anymore.  It's also very possible that I'm misremembering this.